src/snowflake/snowpark/session.py

```diff
@@ -2417,7 +2417,16 @@ def _infer_is_return_table(
         self, sproc_name: str, *args: Any, log_on_exception: bool = False
     ) -> bool:
         func_signature = ""
+        sproc_name = sproc_name.strip().upper()
         try:
+            # do not run describe procedure on system procedures
```

Comment:
NIT: I'm a bit confused with this comment. It implies that this function `_infer_is_return_table` is describing a func while it actually check whether it's a table function or not.

Reply:
this function is called by Session.call which is used to call stored procs. We use this function to check if the return type of the called stored procedure is table or not. Describe works fine for user defined stored procs but raises sql exception as noted by the JIRA.

```
describe procedure SYSTEM$SEND_EMAIL(VARCHAR, VARCHAR, VARCHAR, VARCHAR);

Procedure 'TESTDB_SNOWPARK_PYTHON.TESTSCHEMA_SNOWPARK_PYTHON.SYSTEM$SEND_EMAIL' does not exist or not authorized.
```
Hence we skip describe procedure for these stored procs here.